cdc/log.cc

```diff
@@ -570,6 +570,20 @@ static schema_ptr create_log_schema(const schema& s, std::optional<utils::UUID>
         b.set_uuid(*uuid);
     }
 
+    /**
+     * #10473 - if we are redefining the log table, we need to ensure any dropped
+     * columns are registered in "dropped_columns" table, otherwise clients will not
+     * be able to read data older than now.
+     */
+    if (old) {
+        // not super efficient, but we don't do this often.
+        for (auto& col : old->all_columns()) {
+            if (!b.has_column({col.name(), col.name_as_text() })) {
+                b.without_column(col.name_as_text(), col.type, api::new_timestamp());
```

Comment:
We should use the timestamp provided by `on_before_update_column_family`.

Reply:
Normal "drop column" uses wall clock time for this, separate from overall alter table mutation timestamp, so I think following the lead there is ok.